api/v1alpha1/wasm_types.go

```diff
@@ -0,0 +1,84 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+)
+
+// Wasm defines a wasm extension.
```

Comment:
maybe some comments here around defaults, such as only `V8` type and `HTTP` filter based WASM modules are supported today